File: Cplusplus/LEVEL 1/콜라츠추측.cpp
//22.01.21 콜라츠 추측 by CJHofficial

#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    long long n = num;
    while (n != 1) {
        if (answer > 500) return -1;
        n = (n % 2 == 0) ? (n / 2) : (n * 3 + 1);
        answer += 1;
    }

    return answer;
}

File: Python/level1/콜라츠 추측.py
def solution(num):
    answer = 0
    for index in range(500):
        if num == 1:
            answer = index
            break
        num = num // 2 if num % 2 == 0 else num * 3 + 1
    else:
        answer = -1
    return answer
